### PR TITLE
nitweb: uml graphs route

### PR DESCRIPTION
**Pull request by Morriar**
Allows nitweb to display UML.

Thanks @R4PaSs!

### CODE REVIEW

File: src/web/web_actions.nit [hunk 1]
```diff
@@ -98,6 +99,42 @@ class DocAction
...
+		if mentity isa MClass then
+			var uml = new UMLModel(model, mainmodule)
+			dot = uml.generate_class_uml.write_to_string
+		else if mentity isa MModule then
+			var uml = new UMLModel(model, mentity)
+			dot = uml.generate_package_uml.write_to_string
+		else
+			return render_error(404, "No diagram matching this namespace.")
+		end
+		var view = new HtmlDotPage(uri, dot, "{mentity.html_name}")
```
Comment (privat) [inline src/web/web_actions.nit]: why `"{}"`?

File: src/web/web_views.nit [hunk 2]
```diff
@@ -119,3 +119,41 @@ class HtmlDocPage
...
+		tpl.add render_dot
+		return tpl
+	end
+
+	private fun render_dot: String do
+		var path = out_file
+		var file = new FileWriter.open("{path}.dot")
+		file.write(dot)
+		file.close
+		var proc = new ProcessReader("dot", "-Tsvg", "-Tcmapx", "{path}.dot")
```
Comment (privat) [inline src/web/web_views.nit]: It is funny, you choose to pass your input as a file, but read the output as a stream.

File: src/web/web_views.nit [hunk 3]
```diff
@@ -119,3 +119,41 @@ class HtmlDocPage
...
+		var svg = new Buffer
+		var i = 0
+		while not proc.eof do
+			i += 1
+			if i < 6 then continue # skip dot default header
+			svg.append proc.read_line
+		end
+		proc.close
```
Comment (privat) [inline src/web/web_views.nit]: the Process API is still crappy. you also need to call `wait` after the `close` (or you will get a zombie invasion)

**Comment by privat** [comment]
+1

What about a demo?

**Comment by Morriar** [comment]
Reroll, way better with ProcessDuplex indeed...

Demos:
* Package diagram http://nitweb.nitlanguage.org/uml/core::core::kernel
* Class diagram http://nitweb.nitlanguage.org/uml/core::core::collection::array::Array

**Comment by privat** [comment]
+1 demo are nice and the code seems neater.

**Comment by R4PaSs** [comment]
Yay ! Nituml is finally useful ! :)

:shipit: !

**Pull request merged**